src/main/scala/com/codacy/parsers/implementation/CloverParser.scala

```diff
@@ -0,0 +1,77 @@
+package com.codacy.parsers.implementation
+
+import java.io.File
+
+import com.codacy.api.{CoverageFileReport, CoverageReport}
+import com.codacy.parsers.CoverageParser
+import com.codacy.parsers.util.{TextUtils, XMLoader}
+
+import scala.util.{Failure, Success, Try}
+import scala.xml.{Elem, Node, NodeSeq}
+
+object CloverParser extends CoverageParser {
+  private val CoverageTag = "coverage"
+  private val ProjectTag = "project"
+  private val MetricsTag = "metrics"
+
+  override val name: String = "Clover"
+
+  override def parse(rootProject: File, reportFile: File): Either[String, CoverageReport] = {
+    val report = Try(XMLoader.loadFile(reportFile)) match {
+      case Success(xml) if hasCorrectSchema(xml) =>
+        Right(xml \\ CoverageTag)
+
+      case Success(_) =>
+        Left(s"Invalid report. Could not find tag hierarchy <$CoverageTag> <$ProjectTag> <$MetricsTag> tags.")
+
+      case Failure(ex) =>
+        Left(s"Unparseable report. ${ex.getMessage}")
```

Comment:
You are returning almost the same Left here and after.. Remove duplication if you can.